tensorflow_io/core/kernels/audio_kernels.cc

```diff
@@ -264,6 +346,34 @@ class AudioResampleOp : public OpKernel {
   int64 quality_;
 };
 
+class DecodeAudioOp : public DecodeAudioBaseOp {
+ public:
+  DecodeAudioOp(OpKernelConstruction *context) : DecodeAudioBaseOp(context) {}
+
+  std::unique_ptr<DecodedAudio> decode(StringPiece &data, void *config) {
```

Comment:
Google's style is to return `Status` in return field of the function, and for any values/pointers that needs to be returned as well, they will be placed at the end of the function with `*`, so something like:
```
Status Decode(StringPiece &data, DecodeAudio** audio);
```

Then you could use:
```
DecodeAudio* audio;
Status status = Decode(data, &audio);

std::unique_ptr<DecodedAudio> d;
d.reset(audio);
```

You might also pass unique_ptr as well I think:
```
Status Decode(StringPiece &data, std::unique_ptr<DecodeAudio>* audio);
```

Reply:
Passing a pointer to a unique pointer sounds pretty hacky, do you think that's a good idea? Probably I'd rather go with the first option. Thx for pointing me to the coding style!